advanced-creation.md

```diff
@@ -101,16 +99,59 @@ const unchangedGot = got.create(defaults);
 const settings = {
 	handler: got.defaults.handler,
 	methods: got.defaults.methods,
-	options: {
-		...got.defaults.options,
+	options: got.assignOptions(got.defaults.options, {
 		headers: {
 			unicorn: 'rainbow'
 		}
-	}
+	})
 };
 
 const unicorn = got.create(settings);
 
 // Same as:
 const unicorn = got.extend({headers: {unicorn: 'rainbow'}});
 ```
+
+### Merging instances
+
+#### got.merge(instances, [methods])
+
+Merges many instances into a single one.
```

Comment:
Would suggest we document the merge strategy. Which options are merged, in what order, etc.

Reply:
Done.